packages/mui-base/src/SnackbarUnstyled/SnackbarUnstyled.tsx

```diff
@@ -0,0 +1,236 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { OverridableComponent } from '@mui/types';
+import ClickAwayListener from '../ClickAwayListener';
+import {
+  SnackbarUnstyledOwnerState,
+  SnackbarUnstyledProps,
+  SnackbarUnstyledRootSlotProps,
+  SnackbarUnstyledTransitionSlotProps,
+  SnackbarUnstyledTypeMap,
+} from './SnackbarUnstyled.types';
+import composeClasses from '../composeClasses';
+import { getSnackbarUnstyledUtilityClass } from './snackbarUnstyledClasses';
+import useSnackbar from './useSnackbar';
+import { useSlotProps, WithOptionalOwnerState } from '../utils';
+
+const useUtilityClasses = () => {
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getSnackbarUnstyledUtilityClass, undefined);
+};
+/**
+ *
+ * Demos:
+ *
+ * - [Snackbar](https://mui.com/base/react-snackbar/)
+ *
+ * API:
+ *
+ * - [SnackbarUnstyled API](https://mui.com/base/api/snackbar-unstyled/)
+ */
+const SnackbarUnstyled = React.forwardRef(function SnackbarUnstyled(
+  props: SnackbarUnstyledProps & { component?: React.ElementType },
+  ref: React.ForwardedRef<any>,
+) {
+  const {
+    autoHideDuration = null,
+    children,
+    ClickAwayListenerProps,
+    component,
+    components = {},
+    componentsProps = {},
+    disableWindowBlurListener = false,
+    onBlur,
+    onClose,
+    onFocus,
+    onMouseEnter,
+    onMouseLeave,
+    open,
+    resumeHideDuration,
+    ...other
+  } = props;
+
+  const classes = useUtilityClasses();
+
+  const { getRootProps, getTransitionProps, exited, onClickAway } = useSnackbar({
+    autoHideDuration,
+    disableWindowBlurListener,
+    onClose,
+    open,
+    resumeHideDuration,
+    ref,
+  });
+
+  const ownerState: SnackbarUnstyledOwnerState = {
+    ...props,
+    autoHideDuration,
+    disableWindowBlurListener,
+    exited,
+  };
+
+  const Root = component || components.Root || 'div';
+
+  const TransitionComponent = components.Transition;
+
+  const componentsPropsRootSlotEventCallback = (
```

Comment:
There is a resolveComponentProps function in mui-base/src/utils that does this

Reply:
Used it thanks.